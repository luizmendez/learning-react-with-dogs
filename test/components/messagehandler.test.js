import React from 'react';
import MessageHandler from 'components/messagehandler/messagehandler';

const noop = () => {};

const props = {
    messagesList: [
        {
            message: 'Error uploading dog pic.',
            type: 'error',
            id: 131313
        }
    ]
};

describe('<MessageHandler />', () => {
    it('renders the MessageHandler component', () => {
        const wrapper = shallow(<MessageHandler {...props} />);
        expect(wrapper.find('.message-handler').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a Snackbar child component', () => {
        const messagesList = [
            {
                message: 'Testing an snackbar message component',
                type: 'info',
                id: 131313
            }
        ];
        const wrapper = mount(<MessageHandler messagesList={messagesList} />);
        expect(wrapper.find('.snackbar').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
});
