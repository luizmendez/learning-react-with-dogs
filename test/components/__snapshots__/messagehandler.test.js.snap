// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageHandler /> renders the MessageHandler component 1`] = `
<div
  className="message-handler"
>
  <Snackbar
    autoClose={true}
    hideCloseButton={false}
    id={131313}
    key="131313"
    message="Error uploading dog pic."
    msToClose={3000}
    type="error"
  />
</div>
`;

exports[`<MessageHandler /> should render a Snackbar child component 1`] = `
<MessageHandler
  callback={[Function]}
  messagesList={
    Array [
      Object {
        "id": 131313,
        "message": "Testing an snackbar message component",
        "type": "info",
      },
    ]
  }
>
  <div
    className="message-handler"
  >
    <Snackbar
      autoClose={true}
      callback={[Function]}
      hideCloseButton={false}
      id={131313}
      key="131313"
      message="Testing an snackbar message component"
      msToClose={3000}
      type="info"
    >
      <div
        className="snackbar alert alert-dismissible alert-info"
      >
        Testing an snackbar message component
        <button
          className="close close-snackbar"
          onClick={[Function]}
          type="button"
        >
          Ã—
        </button>
      </div>
    </Snackbar>
  </div>
</MessageHandler>
`;
